### DISCUSSION TITLE
Support for creating Custom Styles / Themes

### DISCUSSION THREAD

**Discussion by YannickMol** [directus/directus#4204] [Feature Requests]
Are there any plans for setting custom styles? Let's say you can upload a `_theme.scss` or `_fonts.scss` file in a`extensions/styles` folder of your Directus project. In these files you can then set the styles you want for your Directus project. I know some of this functionality can be achieved through the Project Settings, but would be nice to extend this to a`styles` folder. 

![image](https://user-images.githubusercontent.com/13256289/108625133-6de39c80-7449-11eb-82f1-1eb9d5683b20.png)

**Comment by philleepflorence** [comment]
I presumed the **Project Settings - CSS Overrides** would be mostly CSS import statements to style files and vendors, but on second look, I see the need for a styles extensions directory for **Full White-Labeling**.

Would the **Project Styles** also be compiled like **Custom Extensions**?

**Reply by rijkvanzanten** [reply]
> I see the need for a styles extensions directory for Full White-Labeling.

I disagree. White-labeling technically is rebranding, which is already possible by just swapping out the logo.

The CSS overrides allow you to inject any CSS, including any of the variables in the light/dark theme, so that should be more than sufficient for white-labeling purposes.

**Reply by philleepflorence** [reply]
> I disagree. White-labeling technically is rebranding, which is already possible by just swapping out the logo.

Curious to see where you got this technical definition as [White-Labeling](https://en.wikipedia.org/wiki/White-label_product) or [Rebranding](https://en.wikipedia.org/wiki/Rebranding) involves more than logo swapping and may mean different things to different organizations.

However, if the only solution is to use the **Project Settings - CSS Overrides**, I can see **CSS import statements** making it more robust.

**Reply by rijkvanzanten** [reply]
From your link:

> A white-label product is a product or service produced by one company (the producer) that other companies (the marketers) rebrand to make it appear as if they had made it.

Directus itself is pretty un-branded, with the default logo/primary accent color being the exceptions. Those can be configured through settings, and thus allow for "full white-labeling". For all other branding (like dedicated fonts/sizing things), the css overrides already provide access to everything you'd need.

**Reply by philleepflorence** [reply]
Like I said, many (marketing) organizations (including 2 branding agencies I occasionally assist) define rebranding differently...for some customizations in D8, there were some extensive style requests beyond what you list.

I am not trying to (re)define or debate white-labeling or rebranding, but saying the solution provided could work.

**Comment by benhaynes** [comment]
Would love to see these be modular (beyond light/dark)...maybe even an extension so that they are more easily shareable. Thanks @YannickMol!

**Comment by tylerforesthauser** [comment]
I’m hoping this discussion is the appropriate place for this but are there plans or have there been considerations made regarding extensibility beyond _adding_ to the available App components? Is it or would it ever be possible to _override_ existing pieces of the Directus App, not only Displays, Interfaces, Modules and Layouts, but even base components, directives and views, for instance?

**Reply by benhaynes** [reply]
I mean, it's open-source, so you can override absolutely anything you want in your fork. As of now, we're not planning to make any other than extensions overridable... but if you have a good 80/20 use case we're happy to discuss! Also, you can re-use the core components (eg: use a Directus checkbox component in an extension), but not sure why they need to be overridden.

**Reply by tylerforesthauser** [reply]
My particular use-case is to completely customize the admin UI to utilize an existing design system. The biggest trouble is that we ended up using a css-in-js solution so simply overriding the CSS isn't enough. I've considered simply forking but I worry it would be too involved to maintain. So I was just curious if was something you've considered before (also it is overlapping functionality with Strapi, so it may have been generally sought after). I'll stew on it more and come back if I've got more universal ideas.

**Reply by emahuni** [reply]
I was hoping the same thing about this framework. That's the only missing piece I can see. I want to be able to "completely customize the admin UI to utilize an existing design system", but maybe have access to certain components from the original app. The idea is to be able to use the components system used by directus to create a totally different looking UI or even website. Instead of creating another app/server for the frontend, why not just use 0.0.0.0:80 as the frontend that is directus hosted and managed and /admin for the admin app and totally customizable in case one needs to change it.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
